src/upnp_xml.cc

```diff
@@ -126,14 +126,20 @@ void UpnpXMLBuilder::renderObject(const std::shared_ptr<CdsObject>& obj, size_t
     result.append_attribute("parentID") = obj->getParentID();
     result.append_attribute("restricted") = obj->isRestricted() ? "1" : "0";
 
-    std::string tmp = obj->getTitle();
-    if ((stringLimit != std::string::npos) && (tmp.length() > stringLimit)) {
-        tmp = tmp.substr(0, getValidUTF8CutPosition(tmp, stringLimit - 3));
-        tmp = tmp + "...";
-    }
+    auto trimString = [stringLimit](const std::string& s) {
+        // Do nothing if limit is negative, or string is already short enough
+        if (stringLimit < 0 || s.length() <= stringLimit)
```

Comment:
size_t cannot be negative.

Reply:
I just matched what `getValidUTF8CutPosition` returns.
https://github.com/gerbera/gerbera/blob/04ce6d5ddebcb6a899b26d154ac792e6ee64be54/src/util/tools.cc#L985

Reply:
Sorry I misunderstood what you were asking for. I see now you mean `stringLimit`. You are correct. I'm surprised the compiler isn't throwing a warning for this.

Reply:
cppcheck does. But yes, I think this is an extremely common pattern such that compilers pretty much have given up.

Reply:
Well I just turned on `-Wall -Wextra -pedantic` to see...

About 370 something warnings. Lots and lots of unused parameters to prune.

Reply:
clang's -Weverything is pretty insane :)

Those unused parameters are questionable. They mostly deal with virtual functions where unused parameters are expected.

Reply:
Yes that makes sense. I wonder if they could be silenced by adding default values. Oh well.